includes/Core/Authentication/Authentication.php

```diff
@@ -456,18 +449,10 @@ private function refresh_auth_token_on_login() {
 
 		$auth_client = $this->get_oauth_client();
 
-		// Initiates Google Client object.
-		$auth_client->get_client();
-
-		// Refresh auth token.
-		$auth_client->refresh_token();
-
-		// If 'invalid_grant' error, disconnect the account.
-		if ( 'invalid_grant' === $this->user_options->get( Clients\OAuth_Client::OPTION_ERROR_CODE ) ) {
-			$this->disconnect();
-
-			// We need to re-set this error so that it is displayed to the user.
-			$this->user_options->set( Clients\OAuth_Client::OPTION_ERROR_CODE, 'invalid_grant' );
+		// Make sure to refresh the access token if necessary.
+		$google_client = $auth_client->get_client();
+		if ( $auth_client->get_access_token() && $google_client->isAccessTokenExpired() ) {
+			$auth_client->refresh_token();
```

Comment:
Why do we need to check if it is expired? Doesn't the Google client refresh it automatically when needed?

Reply:
See above.